### DISCUSSION TITLE
Can I subscribe (realtime) to all collections create/delete/update operations with web API?

### DISCUSSION THREAD

**Discussion by babakfp** [pocketbase/pocketbase#5022] [Q&A] [answered]
Hi 👋

I wrote a bunch of code to read PB collections and generate TS types for it (same as pocketbase-typegen).

I used pb_hooks to generate the TS types when create/delete/update happen on collections.

- `onCollectionAfterCreateRequest`
- `onCollectionAfterUpdateRequest`
- `onCollectionAfterDeleteRequest`

When any of those happen, I send a POST request (with `$http.send`) to my Node.js API, to ask it to run the TS type generation function.

All of this is great, but the `onAfterBootstrap` operation didn't work the same as the above operations. Because when I send the POST request to the Node.js API, the Node.js server is not able to connect to PB.

My workaround for this was to use `await pb.realtime.subscribe("PB_CONNECT")` in my frontend app.

Things work as expected, but it would be easier to have a NPM library that you just import a function, run it, and that's it, rather than doing a bunch of things.

I think this is doable if PB had a real-time feature for the above operations. I would also like to avoid using "Extend with JS" because it's not easy to work with because of the limitations. Maybe having those features as a real-time API could be a good idea? I'm not sure if it's possible or what not.

Thanks.

**Comment by ganigeorgiev** [comment, marked as answer]
No, currently realtime events are fired only for the record related operations.

**Reply by babakfp** [reply]
Are you OK with a PR for this? I don't know how to do it, but maybe someone else is interested to take a look into it.

Only the bare minimum needed. Basically a feature to allow real-time subscriptions on collections create/update/delete operations.

Examples:

```ts
await pb.collections.realtime.subscribe()
```

or

```ts
await pb.realtime.subscribe("collections")
```

**Reply by ganigeorgiev** [reply]
No.
